core/src/main/java/org/ehcache/core/config/ExpiryUtils.java

```diff
@@ -173,5 +177,39 @@ public static long getExpirationMillis(long now, Duration duration) {
     } catch (ArithmeticException e) {
       return Long.MAX_VALUE;
     }
+
+  }
+
+  /**
+   * Returns the expiry for creation duration returned by the provided {@link ExpiryPolicy} but checks for immediate
+   * expiry, null expiry and exceptions. In all those cases, {@code null} will be returned.
+   *
+   * @param key key to pass to {@link ExpiryPolicy#getExpiryForCreation(Object, Object)}
+   * @param value value to pass to to pass to {@link ExpiryPolicy#getExpiryForCreation(Object, Object)}
+   * @param expiry expiry queried
+   * @param <K> type of key
+   * @param <V> type of value
+   * @return the duration returned by to pass to {@link ExpiryPolicy#getExpiryForCreation(Object, Object)}, {@code null}
+   * if the call throws an exception, if the returned duration is {@code null} or if it is lower or equal to 0
+   */
+  public static <K, V> Duration getExpiryForCreation(K key, V value, ExpiryPolicy<? super K, ? super V> expiry) {
+    Duration duration;
+    try {
+      duration = expiry.getExpiryForCreation(key, value);
+    } catch (RuntimeException e) {
+      LOG.error("Expiry computation caused an exception - Expiry duration will be 0", e);
+      return Duration.ZERO;
+    }
+
+    try {
+      if (Duration.ZERO.compareTo(duration) >= 0) {
+        return Duration.ZERO;
+      }
+    } catch(NullPointerException e) {
```

Comment:
I think I'd prefer an explicit null-check here.

Reply:
:-) I thought about it and decided that it is unlikely to be null so this is fine. But as you wish